descqa/DeltaSigmaTest.py

```diff
@@ -101,10 +101,20 @@ def run_on_single_catalog(self, catalog_instance, catalog_name, output_dir):
             fig2 = plt.figure(2, figsize=(12,5))
 
         if self.data == 'sdss_main':
-            res = catalog_instance.get_quantities(['redshift_true', 'ra', 'dec', 'shear_1', 'shear_2',
-                    'mag_true_i_sdss', 'mag_true_z_sdss','mag_true_g_sdss', 'mag_true_r_sdss', 'stellar_mass_bulge', 'stellar_mass_disk','Mag_true_g_sdss_z0','Mag_true_r_sdss_z0'])
+            if 'stellar_mass' in catalog_instance.list_all_quantities():
+                res = catalog_instance.get_quantities(['redshift_true', 'ra', 'dec', 'shear_1', 'shear_2',
+                                                       'mag_true_i_sdss', 'mag_true_z_sdss','mag_true_g_sdss',
+                                                       'mag_true_r_sdss', 'stellar_mass',
+                                                       'Mag_true_g_sdss_z0', 'Mag_true_r_sdss_z0'])
+                sm = res['stellar_mass']
+            else:
+                res = catalog_instance.get_quantities(['redshift_true', 'ra', 'dec', 'shear_1', 'shear_2',
+                                                       'mag_true_i_sdss', 'mag_true_z_sdss','mag_true_g_sdss',
+                                                       'mag_true_r_sdss', 'stellar_mass_bulge', 'stellar_mass_disk',
+                                                       'Mag_true_g_sdss_z0', 'Mag_true_r_sdss_z0'])
+                sm = res['stellar_mass_bulge'] + res['stellar_mass_disk']
+
```

Comment:
```suggestion
            if not catalog_instance.has_quantities(['stellar_mass']):
                catalog_instance.add_derived_quantity('stellar_mass', np.add, 'stellar_mass_bulge', 'stellar_mass_disk')
            res = catalog_instance.get_quantities([
                 'redshift_true', 'ra', 'dec', 'shear_1', 'shear_2', 'stellar_mass',
                 'mag_true_i_sdss', 'mag_true_z_sdss','mag_true_g_sdss', 'mag_true_r_sdss',
                 'Mag_true_g_sdss_z0', 'Mag_true_r_sdss_z0',
             ])
            sm = res['stellar_mass']
```

Use `add_derived_quantity` when `stellar_mass` is not available to simply the logic here.